cmd/kube-apiserver/app/server.go

```diff
@@ -455,6 +456,7 @@ func (s *APIServer) Run(_ []string) error {
 						glog.Infof("Using self-signed cert (%s, %s)", s.TLSCertFile, s.TLSPrivateKeyFile)
 					}
 				}
+				_ = systemd.SdNotify("READY=1\n")
```

Comment:
Looks like the best way to figure out if you are running under systemd is to check the file system to see if the /run/systemd/system/ directory exisits.  See http://www.freedesktop.org/software/systemd/man/sd_booted.html for more details.